src/PolynomialSolutions.jl

```diff
@@ -108,6 +112,32 @@ function multiply_by_r(p::Polynomial{N,T}, k::Int) where {N,T}
     return multiply_by_r(q, k - 2)
 end
 
+"""
+    multiply_by_anisotropic_r(A::SMatrix{N,N,T}, p::Polynomial, k::Int = 2)
+
+Multiply a polynomial `p` by the monomial `r_A^k`, where `r_A = |r^T A^{-1} r|`,
+r = (x_1, x_2, ... x_n]) and `k` is an even positive integer.
+"""
+function multiply_by_anisotropic_r(A::SMatrix{N,N,T}, p::Polynomial{N,T}, k::Int) where {N,T}
+    @assert iseven(k)
+    k == 0 && return p
+    # TODO We can't treat rationals (even GenericLinearAlgebra is missing this?)
+    # because of A^{-1}. What is the most generic thing we should allow?
+    p = convert_coefs(p, Float64)
+    order2coeff = empty(p.order2coeff)
+    invA = LinearAlgebra.inv(A)
+    for (θ, c) in p.order2coeff
+        for i in 1:N
+            for j in 1:N
+                θ′ = ntuple(l -> θ[l] + Int(l == j) + Int(l == i), length(θ))
+                order2coeff[θ′] = get(order2coeff, θ′, zero(T)) + c*invA[i, j]
+            end
+        end
+    end
+    q = Polynomial(order2coeff)
+    return multiply_by_anisotropic_r(A, q, k - 2)
+end
+
```

Comment:
Here, we don't have to give such a strict type to `A`. In fact, instead of having `N` as a parametric type, we can just get it from `N,M = size(A)`, and assert that they are the same. If `A` turns out to be statically sized, then the compiler can infer `N` and optimise a few things, but IMO that is a detail.

Also, do we really want to invert `A` explicitly? I am always scared when I see `inv` 😅 . Probably fine here though...